### PR TITLE
Fix ROC metric for CUDA tensors

### PR DESCRIPTION
**Pull request by tridao**
## What does this PR do?

Previously roc metric (and auroc) errors when passed in CUDA tensors,
due to torch.tensor construction without specifying device.
This fixes the error by using F.pad instead.

# Before submitting

- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [X ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [X] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you create a separate PR for every change.
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? 
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by justusschock** [comment]
@tridao thanks a lot for this fix. You are right, this didn't occur, since we don't have tests with GPU tensors.

**Comment by justusschock** [comment]
@tridao I also added changes to the tests in your PR. I will resolve the pep8 issues, once these changes are passing :)

File: tests/metrics/functional/test_classification.py [hunk 1]
```diff
@@ -47,23 +47,37 @@ def test_against_sklearn(sklearn_metric, torch_metric):
     """Compare PL metrics to sklearn version."""
     pred = torch.randint(10, (500,))
     target = torch.randint(10, (500,))
+    
+    if torch.cuda.is_available():
+        device = 'cuda'
+    else:
+        device = 'cpu'
```
Comment (Borda) [inline tests/metrics/functional/test_classification.py:54]:
```suggestion
    device = 'cuda' if torch.cuda.is_available() else 'cpu'
```

File: tests/metrics/functional/test_classification.py [hunk 2]
```diff
@@ -47,23 +47,37 @@ def test_against_sklearn(sklearn_metric, torch_metric):
...
     assert torch.allclose(
-        torch.tensor(sklearn_metric(target, pred), dtype=torch.float),
+        torch.tensor(sklearn_metric(target, pred), dtype=torch.float, device=device),
         torch_metric(pred, target))
 
     pred = torch.randint(5, (200,))
     target = torch.randint(10, (200,))
+    
+    pred = pred.to(device)
+    target = target.to(device)
```
Comment (Borda) [inline tests/metrics/functional/test_classification.py]:
```suggestion
    pred = torch.randint(5, (200,)).to(device)
    target = torch.randint(10, (200,)).to(device)
```

File: tests/metrics/functional/test_classification.py [hunk 3]
```diff
@@ -47,23 +47,37 @@ def test_against_sklearn(sklearn_metric, torch_metric):
...
     assert torch.allclose(
-        torch.tensor(sklearn_metric(target, pred), dtype=torch.float),
+        torch.tensor(sklearn_metric(target, pred), dtype=torch.float, device=device),
         torch_metric(pred, target))
 
     pred = torch.randint(10, (200,))
     target = torch.randint(5, (200,))
+    
+    pred = pred.to(device)
+    target = target.to(device)
```
Comment (Borda) [inline tests/metrics/functional/test_classification.py]:
```suggestion
    pred = torch.randint(10, (200,)).to(device)
    target = torch.randint(5, (200,)).to(device)
```

File: tests/metrics/functional/test_classification.py [hunk 4]
```diff
@@ -47,23 +47,37 @@ def test_against_sklearn(sklearn_metric, torch_metric):
...
     pred = torch.randint(10, (500,))
     target = torch.randint(10, (500,))
+    
+    if torch.cuda.is_available():
+        device = 'cuda'
+    else:
+        device = 'cpu'
+        
+    pred = pred.to(device)
+    target = target.to(device)
```
Comment (Borda) [inline tests/metrics/functional/test_classification.py]:
```suggestion
    device = 'cuda' if torch.cuda.is_available() else 'cpu'

    pred = torch.randint(10, (500,)).to(device)
    target = torch.randint(10, (500,)).to(device)
```

File: tests/metrics/functional/test_classification.py [hunk 5]
```diff
@@ -45,25 +45,30 @@
 ])
 def test_against_sklearn(sklearn_metric, torch_metric):
     """Compare PL metrics to sklearn version."""
-    pred = torch.randint(10, (500,))
-    target = torch.randint(10, (500,))
+    device = 'cuda' if torch.cuda.is_available() else 'cpu'
```
Comment (williamFalcon) [inline tests/metrics/functional/test_classification.py:48]: the lightning module has a self.device property...

Also, we don't know if user will always be on cuda or not... so this should not depend on that.
Whether a metric needs cuda, tpu or cpu support should come from lightning or a module but not the user.

Comment (Borda) [reply tests/metrics/functional/test_classification.py:48]: this is just for test

**Comment by justusschock** [comment]
@williamFalcon this is just for **testing** the functional interface, where the computation is on the same device as the inputs :)
